src/context/campaignConfigsStore.tsx

```diff
@@ -107,27 +134,80 @@ export const CampaignConfigsStoreContextProvider: FunctionComponent = ({
     setAllConfigs({});
   }, []);
 
-  const [, setState] = useState();
   useEffect(() => {
-    const loadCampaignConfigFromStore = async (): Promise<void> => {
-      const campaignConfigsString = await AsyncStorage.getItem(
-        CAMPAIGN_CONFIGS_STORE_KEY
-      );
-      if (!campaignConfigsString) {
-        setHasLoadedFromStore(true);
-        return;
+    /**
+     * Migrates credentials from old config store to new config store. Clears all old storage locations.
+     * @param newStorageHasData whether most recent data has been found. If this is true, does not
+     * attempt any migration and just clears old storage locations
+     * @returns the updated value of {@link newStorageHasData}, i.e. true if any updated data was found
+     * in the older storage (if {@link newStorageHasData} was passed in as true, it will always return true)
+     */
+    const migrateOldConfigFromStore = async (
+      newStorageHasData: boolean
+    ): Promise<boolean> => {
+      let hasUpdatedData = newStorageHasData;
+      // check v1 storage
+      if (!hasUpdatedData) {
+        const configString = await AsyncStorage.getItem(
+          CAMPAIGN_CONFIGS_STORE_KEY
+        );
+        if (configString) {
+          try {
+            const configStringFromStore: CampaignConfigsMap = JSON.parse(
+              configString
+            );
+            setAllConfigs(configStringFromStore);
+            hasUpdatedData = true;
+          } catch (e) {
+            setState(() => {
+              throw new CampaignConfigError(e);
+            });
+          }
+        }
+      }
+      if (hasUpdatedData) {
+        await AsyncStorage.removeItem(CAMPAIGN_CONFIGS_STORE_KEY);
       }
+
+      return hasUpdatedData;
+    };
+
+    const loadCampaignConfigFromStore = async (): Promise<void> => {
+      let newStorageHasData = false;
```

Comment:
instead of having a variable as toggle, what do you think if we use `campaignConfigsString` instead?
```suggestion
      let campaignConfigsString: WhateverTheTypeIs;
```